kernel/libc/posix/clock_gettime.c

```diff
@@ -1,24 +1,43 @@
 /* KallistiOS ##version##
 
    clock_gettime.c
-   Copyright (C) 2023 Falco Girgis
+   Copyright (C) 2023, 2024 Falco Girgis
 */
 
-#include <time.h>
 #include <arch/timer.h>
 #include <arch/rtc.h>
+
+#include <dc/perfctr.h>
+
+#include <time.h>
 #include <errno.h>
+#include <assert.h>
+#include <stdlib.h>
+
+int clock_getcpuclockid(pid_t pid, clockid_t *clock_id) {
+    /* pid of 0 means the current process,
+       and we only support a single process. */
+    if(pid)
+        return ENOSYS;
```

Comment:
Would it be more appropriate for this to be ESRCH? Based on this: https://pubs.opengroup.org/onlinepubs/009696899/functions/clock_getcpuclockid.html it's what should be returned when you can't find the process passed as pid.

Reply:
Good call!

Reply:
Shit. Okay. Now I remember what the deal was with this... So I was returning `ENOSYS` rather than `ESRCH`, because it looks as though we're returning the thread ID from `getpid()` or whatever standard POSIX process ID function or whatever... So it would be kind of weird to say we couldn't find the pid when they could potentially pass it straight from that function into here... Thought the "No support" error made more sense given that context? Maybe @ljsebald has an opinion on this one? I can see your point too...

Reply:
I agree with @QuzarDC that this should be `ESRCH`.

Reply:
Thank you, both! That's what I was waiting to hear. On it!